src/graphics/canvas-layer.js

```diff
@@ -193,7 +222,7 @@ Crafty.canvasLayerObject = {
             for (j = 0, len = q.length; j < len; ++j) {
                 obj = q[j];
 
-                if (dupes[obj[0]] || !obj._visible || !obj.__c.Canvas)
+                if (dupes[obj[0]] || !obj._visible || (obj._drawLayer !== this) )
```

Comment:
Why is the check `obj._drawLayer !== this` needed here? Could a dirty entity be detached from current layer and attached to other layer before being drawn (and thus undirtied)?